Dockerfile

```
@@ -25,15 +28,36 @@ RUN cargo chef prepare --recipe-path recipe.json
+ENV CARGO_UNSTABLE_BUILD_STD="std,panic_abort"
+
+# The dependency cook runs before the source COPY, so make the nightly
+# override explicit in /app instead of relying on rustup walking up to the
+# chef stage's copy at /; the nightly-only RUSTFLAGS above depends on it.
+COPY rust-toolchain.toml .
 COPY --from=planner /app/recipe.json recipe.json
-RUN cargo chef cook --release --recipe-path recipe.json
+# build-std demands an explicit --target; use the image's own host triple so
+# multi-arch builds (e.g. buildx linux/arm64) keep producing native binaries
+# exactly like the implicit-target build did.
+RUN rustc -vV | sed -n 's/^host: //p' > /rust-target
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)